.github/workflows/release.yml

```diff
@@ -0,0 +1,116 @@
+name: release
+on:
+  push:
+    tags:
+    - "v[0-9]+.[0-9]+.[0-9]+"
+
+jobs:
+  create-release:
+    name: create-release
+    runs-on: ubuntu-latest
+    env:
+      GITU_VERSION: ""
+    outputs:
+      upload_url: ${{ steps.release.outputs.upload_url }}
+      gitu_version: ${{ env.GITU_VERSION }}
+    steps:
+      - name: Get the release version from the tag
+        shell: bash
+        if: env.GITU_VERSION == ''
+        run: |
+          # See: https://github.community/t5/GitHub-Actions/How-to-get-just-the-tag-name/m-p/32167/highlight/true#M1027
+          echo "GITU_VERSION=${GITHUB_REF#refs/tags/}" >> $GITHUB_ENV
+          echo "version is: ${{ env.GITU_VERSION }}"
+      - name: Create GitHub release
+        id: release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ env.GITU_VERSION }}
+          release_name: ${{ env.GITU_VERSION }}
+
+  build-release:
+    name: build-release
+    needs: ['create-release']
+    runs-on: ${{ matrix.os }}
+    env:
+      CARGO: cargo
+      TARGET_FLAGS: "--target ${{ matrix.target }}"
+      TARGET_DIR: ./target/${{ matrix.target }}
+    strategy:
+      matrix:
+        build: [linux, linux-arm, macos]
+        include:
+        - build: linux
+          os: ubuntu-latest
+          rust: stable
+          target: x86_64-unknown-linux-gnu
+        - build: linux-arm
+          os: ubuntu-latest
+          rust: stable
+          target: arm-unknown-linux-gnueabihf
+        - build: macos
+          os: macos-latest
+          rust: stable
+          target: x86_64-apple-darwin
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: Install Rust
+      uses: dtolnay/rust-toolchain@master
+      with:
+        toolchain: ${{ matrix.rust }}
+        target: ${{ matrix.target }}
```

Comment:
cargo / rust is already available when using ubuntu-latest (pre-installed for actions I believe?)

Reply:
This uses a matrix to allow using different versions of rust. I've only added stable binaries but you could easily duplicate the targets to build nightly versions as well.

Reply:
Also I couldn't find any references to cargo or rust in https://github.com/actions/runner-images/blob/main/images/ubuntu/toolsets/toolset-2204.json

Reply:
No biggie, lets stick with it :+1: